appengine-java11/tasks/src/main/java/com/example/task/CreateTask.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.task;
+
+import com.google.cloud.tasks.v2beta3.AppEngineHttpRequest;
+import com.google.cloud.tasks.v2beta3.CloudTasksClient;
+import com.google.cloud.tasks.v2beta3.HttpMethod;
+import com.google.cloud.tasks.v2beta3.QueueName;
+import com.google.cloud.tasks.v2beta3.Task;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Timestamp;
+
+import java.nio.charset.Charset;
+import java.time.Clock;
+import java.time.Instant;
+
+import org.apache.commons.cli.CommandLine;
```

Comment:
There has been a bunch of pushback on adding CLI's to samples (the new sample format mandates you don't unless it's an explicit requirement for the sample). This is because a) it ends up being a bunch of boilerplate code, b) everyone uses a different cli framework and so it becomes a pain to maintain or update future versions. (Java is also particularly painful since it's hard to remember the correct execution syntax.)

A better alternative would be to add an HTTP endpoint that covers the task creation instead, and provide the user with instructions using `curl`.

Reply:
Removed the larger CLI; however still using main in order to be executable.